lib/core/Core.ts

```diff
@@ -65,7 +66,14 @@ export default class Core {
    * The initialization method that must be called before consumption of this core object.
    * The method starts the Observer and Batch Writer.
    */
-  public async initialize () {
+  public async initialize (customLogger?: ISidetreeLogger) {
```

Comment:
Ali suggested using singleton and delegating functionality to improve type safety. Let's talk about it. I think it's not a bad idea.

Reply:
Took Ali's suggestion.